walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,75 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> possibleTimes = new ArrayList<>();
+    List<TimeRange> conflictingTimes = new ArrayList<>();
+    Collection<String> attendees = request.getAttendees();
+
+    // Find every TimeRange that needs to be avoided
+    for (Event event : events) {
+      for (String attendee : event.getAttendees()) {
+        if (attendees.contains(attendee)) {
+          conflictingTimes.add(event.getWhen());
+          break;
+        }
+      }
+    }
+
+    Collections.sort(conflictingTimes, TimeRange.ORDER_BY_START);
+
+    // Remove duplicates and overlaps
+    for (int i = 0; i < conflictingTimes.size() - 1; i++) {
+      TimeRange first = conflictingTimes.get(i);
+      TimeRange second = conflictingTimes.get(i + 1);
+
+      if (first.overlaps(second)) {
+        if (!first.contains(second)) {
+          TimeRange combined = TimeRange.fromStartEnd(first.start(), second.end(), false);
+          conflictingTimes.set(i, combined);
+        }
+
+        conflictingTimes.remove(second);
+        i--;
+      }
+    }
```

Comment:
I think trying to mutate a list that was being iterated on requires extra attention and I would probably try to avoid altogether to avoid potential errors/issues. Instead I would opt to create another list.